cmd/timestamp-server/app/serve.go

```diff
@@ -80,6 +81,31 @@ var serveCmd = &cobra.Command{
 		port := int(viper.GetUint("port"))
 		scheme := viper.GetStringSlice("scheme")
 
+		ntpMonitoring := viper.GetString("ntp-monitoring")
+		if ntpMonitoring != "" {
+			log.Logger.Infof("ntp monitoring: %s", ntpMonitoring)
+			go func() {
+				ntpm, err := ntpmonitor.New(ntpMonitoring)
+				if err != nil {
+					log.Logger.Fatalf("error initializing ntp monitor %s", err)
+				}
+
+				// Restart ntp monitor if needed
+				for {
+					err = ntpm.Start()
+					if err != nil {
+						log.Logger.Warnf("ntp monitor: %s", err)
+					}
+					// Local time have drifted from the
+					// observed NTP time. Shut down the
+					// service.
+					if err == ntpmonitor.ErrInvTime {
+						log.Logger.Fatalf("local time can not be trusted: %s", err)
```

Comment:
Can we record a metric rather than crash? I'd worry some brief outage or misconfiguration in an NTP server would result in a sigstore outage.